### PR TITLE
En_Tsn

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Tsn/z_en_tsn.c [hunk 1]
```diff
@@ -28,58 +40,598 @@ const ActorInit En_Tsn_InitVars = {
...
+
+    if (Message_GetState(&globalCtx->msgCtx) == 0x10) {
+        sp24 = func_80123810(globalCtx);
+        if (sp24 != 0) {
+            gSaveContext.weekEventReg[26] |= 2;
+        }
+
+        if (sp24 > 0) {
+            sp24 = sp24;
+            func_801477B4(globalCtx);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Tsn/z_en_tsn.c:324]:
```suggestion
        if (sp24 > 0) {
            func_801477B4(globalCtx);
```
Is this needed?

File: src/overlays/actors/ovl_En_Tsn/z_en_tsn.c [hunk 2]
```diff
@@ -28,58 +40,598 @@ const ActorInit En_Tsn_InitVars = {
...
+            this->actionFunc = func_80AE0704;
+            if (sp24 == 19) {
+                if (CHECK_QUEST_ITEM(QUEST_UNK_19)) {
+                    if (func_8013A4C4(9)) {
+                        player->actor.textId = 0x107B;
+                        return;
+                    }
+
+                    if (func_8013A4C4(11)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Tsn/z_en_tsn.c:333]:
```suggestion
                    if (func_8013A4C4(0xB)) {
```
This is a flag so better to have it in hex

**Review by jpburnett — APPROVED** [review]
catcha da feesh 🎣

File: src/overlays/actors/ovl_En_Tsn/z_en_tsn.c [hunk 3]
```diff
@@ -28,58 +40,597 @@ const ActorInit En_Tsn_InitVars = {
...
+        } else {
+            Actor_MarkForDeath(&this->actor);
+        }
+        return;
+    }
+
+    this->unk_1D8 = func_80ADFCA0(globalCtx);
+    this->unk_220 = 0;
+
+    if (this->unk_1D8 == 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Tsn/z_en_tsn.c:117]:
```suggestion
    if (this->unk_1D8 == NULL) {
```

**Review by AngheloAlf — APPROVED** [review]

**Comment by AngheloAlf** [comment]
This PR has conflicts

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**